### DISCUSSION TITLE
How to enable `redirect` with `type` push in next-intl.

### DISCUSSION THREAD

**Discussion by siddhart1o1** [amannn/next-intl#472] [Q&A] [answered]
I am using next-intl with Next.js 13 in server actions, and I want to `redirect` the user to another page using `redirect from next-intl/server`. However, I want to maintain browser history. Unfortunately, the redirect function in this library does not have a type option to specify whether it should be a push or replace action. How can I achieve a push-type redirect using this library on the server side?

**Comment by amannn** [comment, marked as answer]
Hey, thanks for reaching out! I noticed this too lately that apparently in a recent Next.js release `redirect` got a second argument. I already have this fixed in https://github.com/amannn/next-intl/pull/426, which I'm planning to roll out soon to the RSC beta. You can watch the PR to stay in the loop!

**Reply by siddhart1o1** [reply]
Is this ready for use? PR is merged I think.

**Reply by amannn** [reply]
Yep, it's currently available in the RSC beta! I might upstream it to stable in the coming weeks along with some other improvements.

**Reply by siddhart1o1** [reply]
how can I use RSC beta?

**Reply by amannn** [reply]
Please see here: https://next-intl-docs.vercel.app/docs/getting-started/app-router-server-components

**Reply by siddhart1o1** [reply]
Okay, I tried it, but it should push the page into the browser history if I use the 'push' type. This way, I can use the browser's back and forward buttons. I am using the redirect function in the server action to redirect with the 'push' type, but the page is not getting pushed to the browser's history stack?

**Reply by amannn** [reply]
I haven't experimented with Server Actions yet, as far as I know, they have experimental support currently. No idea if Next.js internally handles these differences correctly yet. `next-intl` only provides a minimal wrapper around what Next.js offers.
